core/input/input_event.cpp

```diff
@@ -107,6 +107,17 @@ bool InputEvent::is_action_type() const {
 	return false;
 }
 
+void InputEvent::copy_metadata(const Ref<InputEvent> &target) const {
```

Comment:
Can you please check, if you could reuse `Object::merge_meta_from`?

https://github.com/godotengine/godot/blob/1f47e4c4e3a09a422e96880a7918d986dd575a63/core/object/object.cpp#L1028-L1034